drivers/stepper/gpio_stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_stepper
+
+#include "zephyr/drivers/gpio.h"
+#include <zephyr/kernel.h>
+#include "zephyr/drivers/stepper.h"
+#include <zephyr/logging/log.h>
+#include "stdlib.h"
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+struct gpio_stepper_motor_controller_data {
+	struct k_work_delayable stepper_dwork;
+	const struct gpio_dt_spec *gpio_in_pins;
+	int32_t min_position;
+	int32_t max_position;
+	int32_t actual_position;
+	int32_t target_position;
+	uint32_t actual_speed;
+	uint32_t delay_in_us;
+	bool is_moving;
```

Comment:
These values can be accessed concurrently by the work queue thread and any thread calling the API implementations, they need to be protected by a spinlock or similar.